packages/ai/src/core.ts

```diff
@@ -111,4 +111,52 @@ export class EdgeDBAI {
 
     return data.response;
   }
+
+  async *getRagAsyncGenerator(
+    message: string,
+    context: QueryContext = this.context
+  ): AsyncGenerator<string, void, undefined> {
+    const response = await this.fetchRag({
+      model: this.options.model,
+      prompt: this.options.prompt,
+      context,
+      query: message,
+      stream: true,
+    });
+
+    if (response.body) {
+      const reader = response.body.getReader();
+      try {
+        while (true) {
+          const { done, value } = await reader.read();
+          if (done) break;
+          yield new TextDecoder().decode(value);
+        }
+      } finally {
+        reader.releaseLock();
+      }
+    }
+  }
+
+  streamRag(message: string, context: QueryContext = this.context): Response {
+    const generator = this.getRagAsyncGenerator(message, context);
+
+    const stream = new ReadableStream<string>({
+      async start(controller: ReadableStreamDefaultController<string>) {
+        try {
+          for await (const chunk of generator) {
+            controller.enqueue(chunk);
+          }
+        } catch (error) {
+          controller.error(error);
+        } finally {
+          controller.close();
+        }
+      },
+    });
```

Comment:
Why do we take the `ReadableStream` from the fetch response body, convert it into an async iterator, then wrap it into a `ReadableStream` again for the output response body? If the idea is to pass the streaming response on to the client, couldn't we just use the fetch response body in the output response body?